include/RAJA/policy/openmp/policy.hpp

```diff
@@ -66,90 +102,96 @@ namespace RAJA
-    : public RAJA::make_policy_pattern<RAJA::Policy::openmp,
-                                       RAJA::Pattern::forall> {
+struct omp_parallel_for_exec : omp_parallel_exec<omp_for_exec> {
 };
 
-template <size_t ChunkSize>
-struct omp_parallel_for_static
-    : public omp_parallel_exec<omp_for_static<ChunkSize>> {
+template <unsigned int N>
+struct omp_parallel_for_static : omp_parallel_exec<omp_for_static<N>> {
 };
 
-struct omp_for_nowait_exec
-    : public RAJA::make_policy_pattern<RAJA::Policy::openmp,
-                                       RAJA::Pattern::forall> {
+///
+/// Policies for applying OpenMP clauses in forallN loop nests.
+///
+struct omp_collapse_nowait_exec
+    : make_policy_pattern_t<Policy::openmp, Pattern::forall, omp::Collapse> {
 };
 
 #if defined(RAJA_ENABLE_TARGET_OPENMP)
 template <size_t Teams>
 struct omp_target_parallel_for_exec
-    : public RAJA::make_policy_pattern<RAJA::Policy::target_openmp,
-                                       RAJA::Pattern::forall> {
+    : make_policy_pattern_t<Policy::target_openmp, Pattern::forall, omp::Target, omp::Teams<Teams>, omp::Distribute> {
 };
 #endif
 
 ///
 /// Index set segment iteration policies
 ///
-struct omp_parallel_for_segit : public omp_parallel_for_exec {
+
+struct omp_parallel_for_segit : omp_parallel_for_exec {
 };
-struct omp_parallel_segit : public omp_parallel_for_segit {
+
+struct omp_parallel_segit : omp_parallel_for_segit {
 };
+
 struct omp_taskgraph_segit
-    : public RAJA::make_policy_pattern<RAJA::Policy::openmp,
-                                       RAJA::Pattern::taskgraph> {
+    : make_policy_pattern_t<Policy::openmp, Pattern::taskgraph, omp::Parallel> {
 };
+
 struct omp_taskgraph_interval_segit
-    : public RAJA::make_policy_pattern<RAJA::Policy::openmp,
-                                       RAJA::Pattern::taskgraph> {
+    : make_policy_pattern_t<Policy::openmp, Pattern::taskgraph, omp::Parallel> {
 };
 
-
-///
-/// Policies for applying OpenMP clauses in forallN loop nests.
-///
-struct omp_collapse_nowait_exec : 
-  public RAJA::make_policy_pattern<RAJA::Policy::openmp, RAJA::Pattern::forall> {};
-
 ///
 ///////////////////////////////////////////////////////////////////////
 ///
 /// Reduction execution policies
 ///
 ///////////////////////////////////////////////////////////////////////
 ///
-struct omp_reduce : public RAJA::make_policy_pattern<RAJA::Policy::openmp,
-                                                     RAJA::Pattern::reduce> {
+
+struct omp_reduce : make_policy_pattern_t<Policy::openmp, Pattern::reduce> {
 };
 
 #if defined(RAJA_ENABLE_TARGET_OPENMP)
 template <size_t Teams>
 struct omp_target_reduce
-    : public RAJA::make_policy_pattern<RAJA::Policy::target_openmp,
-                                       RAJA::Pattern::reduce> {
+    : make_policy_pattern_t<Policy::target_openmp, Pattern::reduce> {
 };
 #endif
 
 struct omp_reduce_ordered : public omp_reduce {
 };
 
+struct omp_reduce_ordered
```

Comment:
I know... I did the merge through the github website and have since fixed it. Not to mention NVCC barfs really badly